src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1075,7 +1079,6 @@ void EmitAllBranches()
                             {
                                 writer.WriteLine($"case {i}: goto {labelMap[i]};");
                             }
-                            writer.WriteLine("default: global::System.Diagnostics.Debug.Fail(\"Unknown backtracking location\"); break;");
```

Comment:
NIT: wouldn't it be good to keep this one in case of failure?

Reply:
This is the only assert in the generated code.  I'm ok keeping it if we think it's helpful in finding source gen bugs, but then I'd also want to go add asserts in a bunch of other places.

Reply:
If it's the only remaining one I agree it's good to remove and then if we later want to find bugs we can add Asserts in all of the places were it can be useful.